client/packages/programs/src/JsonForms/common/JsonForm.tsx

```diff
@@ -154,8 +161,18 @@ const FormComponent = ({
   // This allows "default" values to be set in the JSON schema
   const handleDefaultsAjv = createAjv({ useDefaults: true });
 
+  const commonKeys = useMemo(() => new Set(Object.keys(common)), []);
```

Comment:
memoized keys in case checking Object.keys repeatedly is expensive. Unsure if it would be for how many keys we have.